presto-hive/src/main/java/com/facebook/presto/hive/StaticHiveCluster.java

```diff
@@ -14,35 +14,82 @@
 package com.facebook.presto.hive;
 
 import com.google.common.net.HostAndPort;
+import io.airlift.log.Logger;
 import org.apache.thrift.transport.TTransportException;
 
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ThreadLocalRandom;
+
+import static com.facebook.presto.hive.RetryDriver.retry;
 import static com.google.common.base.Preconditions.checkArgument;
 import static com.google.common.base.Preconditions.checkNotNull;
 
 public class StaticHiveCluster
         implements HiveCluster
 {
-    private final HostAndPort address;
+    private static final Logger log = Logger.get(StaticHiveCluster.class);
+    private final List<HostAndPort> addresses;
     private final HiveMetastoreClientFactory clientFactory;
 
-    public StaticHiveCluster(HostAndPort address, HiveMetastoreClientFactory clientFactory)
+    public StaticHiveCluster(List<HostAndPort> addresses, HiveMetastoreClientFactory clientFactory)
     {
-        checkNotNull(address, "address is null");
-        checkArgument(address.hasPort(), "address does not have a port");
+        checkNotNull(addresses, "addresses is null");
+        checkArgument(addresses.size() > 0, "must have one or more addresses");
+        addresses.forEach(StaticHiveCluster::checkPort);
```

Comment:
Actually, just use a normal for-each loop. We normally only use `.forEach()` when it is the result of a stream, etc.